iib/web/models.py

```
@@ -1075,6 +1095,170 @@ def get_mutable_keys(self):
+    index_image = db.relationship('Image', foreign_keys=[index_image_id], uselist=False)
+
+    source_from_index_id = db.Column(db.Integer, db.ForeignKey('image.id'), nullable=False)
+    source_from_index_resolved_id = db.Column(db.Integer, db.ForeignKey('image.id'))
+    source_from_index = db.relationship('Image', foreign_keys=[source_from_index_id], uselist=False)
+    source_from_index_resolved = db.relationship(
+        'Image', foreign_keys=[source_from_index_resolved_id], uselist=False
+    )
+
+    target_index_id = db.Column(db.Integer, db.ForeignKey('image.id'), nullable=True)
+    target_index_resolved_id = db.Column(db.Integer, db.ForeignKey('image.id'))
+    target_index = db.relationship('Image', foreign_keys=[target_index_id], uselist=False)
+    target_index_resolved = db.relationship(
+        'Image', foreign_keys=[target_index_resolved_id], uselist=False
+    )
+
+    __mapper_args__ = {
+        'polymorphic_identity': RequestTypeMapping.__members__['merge_index_image'].value,
+    }
+
+    @classmethod
+    def from_json(cls, kwargs, batch=None):
+        """
+        Handle JSON requests for the merge-index-image API endpoint.
+
+        :param dict kwargs: the JSON payload of the request.
+        :param Batch batch: the batch to specify with the request.
+        """
+        request_kwargs = deepcopy(kwargs)
+
+        deprecation_list = request_kwargs.pop('deprecation_list', [])
+        if not isinstance(deprecation_list, list) or any(
+            not item or not isinstance(item, str) for item in deprecation_list
+        ):
+            raise ValidationError(
+                '"deprecation_list" should be an empty array or an array of strings.'
+            )
+
+        request_kwargs['deprecation_list'] = [
+            Image.get_or_create(pull_specification=item) for item in deprecation_list
+        ]
+
+        source_from_index = request_kwargs.pop('source_from_index', None)
+        if not isinstance(source_from_index, str):
```

```suggestion
        if not (isinstance(source_from_index, str) and source_from_index):
```

---

`source_from_index` is a required parameter. I think it should throw an error if it's None too, no?

---

This suggestion does that. It ensures it is a string with at least a character in it.